src/Core/Domain/Shop/ValueObject/ShopConstraint.php

```diff
@@ -52,29 +52,29 @@ class ShopConstraint
     /**
      * Constraint to get configuration for a specific shop
      *
-     * @param int $shopId
+     * @param int|null $shopId
      * @param bool $isStrict
      *
      * @return static
      *
      * @throws ShopException
      */
-    public static function shop(int $shopId, bool $isStrict = false): self
+    public static function shop(?int $shopId, bool $isStrict = false): self
     {
         return new static($shopId, null, $isStrict);
     }
 
     /**
      * Constraint to get configuration for a specific shop group
      *
-     * @param int $shopGroupId
+     * @param int|null $shopGroupId
      * @param bool $isStrict
      *
      * @return static
      *
      * @throws ShopException
      */
-    public static function shopGroup(int $shopGroupId, bool $isStrict = false): self
+    public static function shopGroup(?int $shopGroupId, bool $isStrict = false): self
```

Comment:
Same here, you shouldn't be able to set a shop group constraint without shopGroupId